src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/Pipeline.cs

```diff
@@ -0,0 +1,498 @@
+
+            if (newDataCreated == null)
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "Event Handler is invalid");
+
+            SinkCallbackEvent sce;
+            if (_sinkEventList.ContainsKey(sinkNodeName) == true)
+            {
+                sce = _sinkEventList[sinkNodeName];
+            }
+            else
+            {
+                sce = new SinkCallbackEvent(sinkNodeName, _handle);
+                _sinkEventList.Add(sinkNodeName, sce);
+            }
+            sce.NewDataCreated += newDataCreated;
+        }
+
+        /// <summary>
+        /// Unregisters new data event handler from sink node.
+        /// </summary>
+        /// <param name="sinkNodeName">The name of sink node</param>
+        /// <param name="newDataCreated">Event handler to be unregistered</param>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed to unregister sink callback because of internal error.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void UnregisterSinkCallback(string sinkNodeName, EventHandler<NewDataEventArgs> newDataCreated)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            /* Check the argument */
+            if (string.IsNullOrEmpty(sinkNodeName) || (_sinkEventList.ContainsKey(sinkNodeName) != true))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "Node Name is invalid");
+
+            if (newDataCreated == null)
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "Event Handler is invalid");
+
+            SinkCallbackEvent sce = _sinkEventList[sinkNodeName];
+            sce.NewDataCreated -= newDataCreated;
+        }
+
+        /// <summary>
+        /// Adds an input data frame to source node.
+        /// </summary>
+        /// <param name="srcNodeName">The name of source node</param>
+        /// <param name="data">TensorsData to be inputed</param>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed to input TensorsData because of internal error.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void InputData(string srcNodeName, TensorsData data)
```

Comment:
How about rename to `AddInputData`?
```suggestion
        public void AddInputData(string srcNodeName, TensorsData data)
```

Reply:
will be updated.